import pandas as pd
from sqlalchemy import create_engine, text

# 1Ô∏è‚É£ Charger le fichier CSV
def load_csv(file_path):
    df = pd.read_csv(file_path, delimiter=';', encoding='utf-8')
    df.columns = df.columns.str.strip().str.replace(" ", "_")
    

    print("Aper√ßu des donn√©es :")
    print(df.head())
    return df

# 2Ô∏è‚É£ Connexion √† MySQL
def connect_to_db():    
    engine = create_engine(f"mysql+mysqlconnector://{'root'}:{''}@{'localhost'}/{'wildlens'}")
    print("‚úÖ Connexion r√©ussie √† la base de donn√©es !")
    return engine

# 3Ô∏è‚É£ Ins√©rer les donn√©es dans MySQL
def insert_data(df, engine, table_name="Animal"):
    try:
        # First truncate the photo table to remove foreign key constraints
        with engine.connect() as conn:
            conn.execute(text("SET FOREIGN_KEY_CHECKS=0"))
            conn.execute(text("TRUNCATE TABLE photo"))
            conn.execute(text("TRUNCATE TABLE animal"))
            conn.execute(text("SET FOREIGN_KEY_CHECKS=1"))
            conn.commit()
            
        # Now we can safely insert the data
        df.to_sql('animal', con=engine, if_exists='append', index=False)
        print(f"‚úÖ Donn√©es ins√©r√©es dans la table {table_name}!")
    except Exception as e:
        print(f"‚ùå Erreur d'insertion : {e}")



# üöÄ Ex√©cution des √©tapes
csv_file = "downloaded_data/infos_especes.csv"  # Remplace par ton chemin correct
df = load_csv(csv_file)
engine = connect_to_db()
insert_data(df, engine)